subxt/src/client.rs

```diff
@@ -264,21 +294,63 @@ where
                 .system_account_next_index(signer.account_id())
                 .await?
         };
-        let call = self
-            .client
-            .metadata()
-            .pallet(C::PALLET)
-            .and_then(|pallet| pallet.encode_call(&self.call))?;
-
-        let signed = extrinsic::create_signed(
-            &self.client.runtime_version,
-            self.client.genesis_hash,
+
+        // 2. SCALE encode call data to bytes (pallet u8, call u8, call params).
+        let call_data = {
+            let mut bytes = Vec::new();
+            let pallet = self.client.metadata().pallet(C::PALLET)?;
+            bytes.push(pallet.index());
+            bytes.push(pallet.call_index::<C>()?);
+            self.call.encode_to(&mut bytes);
+            Encoded(bytes)
+        };
+
+        // 3. construct our custom additional/extra params.
+        let additional_and_extra_params = X::new(
+            self.client.runtime_version.spec_version,
+            self.client.runtime_version.transaction_version,
             account_nonce,
-            call,
-            signer,
-            additional_params,
-        )
-        .await?;
-        Ok(signed)
+            self.client.genesis_hash,
+            other_params,
+        );
+
+        // 4. construct signature
+        let signature = {
+            let mut bytes = Vec::new();
+            call_data.encode_to(&mut bytes);
+            additional_and_extra_params.encode_extra_to(&mut bytes);
+            additional_and_extra_params.encode_additional_to(&mut bytes);
+            if bytes.len() > 256 {
+                signer.sign(&sp_core::blake2_256(&bytes))
+            } else {
+                signer.sign(&bytes)
+            }
+        };
+
+        // 5. Encode extrinsic, now that we have the parts we need.
+        // (this may not be 100% correct but should be close enough for this example)
```

Comment:
I assume this comment is a copy/paste error, because it should be 100% correct. 

We should replace it with an explanation that this is a compatible copy of the `Encode` implementation for `UncheckedExtrinsic` for version 4.

Reply:
Ah, indeed, it was copied from my original example :) I'll fix up the comments!